src/java.base/share/classes/java/io/Console.java

```diff
@@ -59,6 +59,12 @@
  * on the objects returned by {@link #reader()} and {@link #writer()} may
  * block in multithreaded scenarios.
  * <p>
+ * Read and write operations use the {@code Charset} returned by the
+ * {@link #charset()} method, unless {@link System##stdin.encoding
+ * stdin.encoding} differs from {@link System##stdout.encoding
+ * stdout.encoding}, in which case read operations use the {@code Charset}
+ * designated by {@code stdin.encoding}.
+ * <p>
```

Comment:
`Console.charset()` states "The returned charset is used for interpreting the input and output source (e.g., keyboard and/or display) specified by the host environment or user, which defaults to the one based on stdout.encoding." If _stdin.encoding_ is set otherwise, this is no longer true, so I think this method may need a wording update as well.

Reply:
Good point. Brought the same wording to the `charset()` method description for further clarification.

Reply:
I think I agree with Stuart and it would be better to say that stdin.encoding is used for reading, and stdout.encoding for writing. They are usually the same but if they differ then Console will return the charset for output.